models/flow/d3q27_cumulant/Dynamics.c.Rt

```diff
@@ -27,19 +27,32 @@ CudaDeviceFunction real_t getRho(){
 CudaDeviceFunction real_t getP(){
 	return ((<?R C(sum(f)) ?>)-1.0)/3.0;
 }
+CudaDeviceFunction vector_t getU_(){
+        real_t d = getRho();
+        vector_t u;
+<?R C(PV(c("u.x","u.y", "u.z")), f %*% U) ?>
+    if((NodeType & NODE_BOUNDARY) == NODE_Wall){
+        u.x = 0;
+        u.y = 0;
+        u.z = 0;
+        }
+        else{
+                u.x = (u.x + ForceX/2.)/d;
+        u.y = (u.y + ForceY/2.)/d;
+        u.z = (u.z + ForceZ/2.)/d;
+    }
+        return u;
+}
 
 CudaDeviceFunction vector_t getU(){
-	real_t d = getRho();
-	vector_t u;
-<?R C(PV(c("u.x","u.y", "u.z")), f %*% U) ?>
-	u.x = (u.x + ForceX/2.)/d;
-	u.y = (u.y + ForceY/2.)/d;
-	u.z = (u.z + ForceZ/2.)/d;
-	return u;
+	RunBoundaries();
+	return getU_();
+
 }
-#ifdef avgUX
+
+#ifdef OPT_AVG
```

Comment:
Aha! According to documentation, it should be "OPTIONS_AVG" not "OPT_AVG"

http://cfd-go.github.io/TCLB/model_info.html
https://github.com/CFD-GO/TCLB/pull/131

So, that was the culprit.